src/rosdep2/main.py

```diff
@@ -368,6 +368,11 @@ def _rosdep_main(args):
                       help="Affects the 'update' verb. "
                            'If specified end-of-life distros are being '
                            'fetched too.')
+    parser.add_option('-t', '--dependency-types', dest='dependency_types',
+                      type="choice", choices=("build", "buildtool", "build_export", "exec", "run", "test", "doc"),
```

Comment:
should this be `nargs='+'` - which should let you omit the `default=[]` because it will be an empty list when not specified. I don't think that `type="choice"` is correct - shouldn't it be `type=str` (which I think is the default)?

Additionally (just thinking out loud) - is there any place that we could query these choices from rather than hardcoding them here? May be better to have a single source of truth for the types of dependencies that are allowed, and if it ever changes this code wouldn't rot.

Reply:
You are right. It is better to have these names defined in a single place. Do you have a suggestion on where they should be?

Reply:
The source of truth seems to be `catking_pkg.package.Package` (https://github.com/ros-infrastructure/catkin_pkg/blob/master/src/catkin_pkg/package.py#L53) - which has the attribute `__slots__`, that might be the best place to parse these values from, though it may not be best practice to go grabbing dunder variables. I'm not super experienced with the source of this catkin infra, input from a maintainer would be helpful on the best approach.